lib/dynamic_config/dcdo.rb

```diff
@@ -9,13 +9,6 @@
 require 'dynamic_config/adapters/memory_adapter'
 
 class DCDOBase < DynamicConfigBase
-  # Adds a listener whose on_change() method will be invoked at least
-  # once whenever the configuration changes. The on_change() method
-  # will be invoked on an arbitrary thread and must not block.
-  def add_change_listener(listener)
-    @datastore_cache.add_change_listener(listener)
-  end
-
```

Comment:
I searched and couldn't find any usage of this either, so looks safe to remove to me, and I appreciate the reduction in test complexity.